syft/pkg/cataloger/java/archive_parser.go

```diff
@@ -272,6 +288,10 @@ func (j *archiveParser) guessMainPackageNameAndVersionFromPomInfo() (name, versi
 	if version == "" && pomProjectObject.JavaPomProject != nil {
 		version = pomProjectObject.Version
 	}
+	if !reflect.ValueOf(pomProjectObject).IsZero() && j.cfg.SearchMavevForLicenses {
```

Comment:
nit: instead of the reflect zero value would it be possible to change the stack variables to be pointers to be able to discern optionality?

Reply:
Converted this to a pointer - I think this is why the integration tests are now failing I have to look at this.